nilearn/decomposition/base.py

```diff
@@ -0,0 +1,466 @@
+                 standardize=True, detrend=True,
+                 low_pass=None, high_pass=None, t_r=None,
+                 target_affine=None, target_shape=None,
+                 mask_strategy='epi', mask_args=None,
+                 memory=Memory(cachedir=None), memory_level=0,
+                 n_jobs=1,
+                 verbose=0):
+        self.n_components = n_components
+        self.random_state = random_state
+        self.mask = mask
+
+        self.smoothing_fwhm = smoothing_fwhm
+        self.standardize = standardize
+        self.detrend = detrend
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.mask_strategy = mask_strategy
+        self.mask_args = mask_args
+        self.memory = memory
+        self.memory_level = memory_level
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+
+    def fit(self, imgs, y=None, confounds=None):
+        """Base fit for decomposition estimators : compute the embedded masker
+
+        Parameters
+        ----------
+        imgs: list of Niimg-like objects
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+            Data on which the mask is calculated. If this is a list,
+            the affine is considered the same for all.
+        """
+        if hasattr(imgs, '__iter__') and len(imgs) == 0:
+            # Common error that arises from a null glob. Capture
+            # it early and raise a helpful message
+            raise ValueError('Need one or more Niimg-like objects as input, '
+                             'an empty list was given.')
+
+        self.masker_ = check_embedded_nifti_masker(self)
+
+        # Avoid warning with imgs != None
+        # if masker_ has been provided a mask_img
+        if self.masker_.mask_img is None:
+            self.masker_.fit(imgs)
+        else:
+            self.masker_.fit()
+        self.mask_img_ = self.masker_.mask_img_
+
+        return self
+
+    def _check_components_(self):
+        if not hasattr(self, 'components_'):
+            if self.__class__.__name__ == 'BaseDecomposition':
+                raise ValueError("Object has no components_ attribute. "
+                                 "This may be because "
+                                 "BaseDecomposition is directly "
+                                 "being used.")
+            else:
+                raise ValueError("Object has no components_ attribute. "
+                                 "This is probably because fit has not "
+                                 "been called.")
+
+    def transform(self, imgs, confounds=None):
+        """Project the data into a reduced representation
+
+        Parameters
+        ----------
+        imgs: iterable of Niimg-like objects
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
```

Comment:
wrong link, should be:
http://nilearn.github.io/manipulating_visualizing/manipulating_images.html#niimg